File: examples/java/core/shared/AppBaseContext.java
package core.shared;

import java.util.LinkedHashMap;
import java.util.Map;

public class AppBaseContext {
  public interface AppLogger {
    void debug(String message, Map<String, Object> meta);
    void info(String message, Map<String, Object> meta);
    void warn(String message, Map<String, Object> meta);
    void error(String message, Map<String, Object> meta);
  }

  public String correlationId;
  public String executionId;
  public String tenantId;
  public String userId;
  public AppLogger logger;
  public Map<String, Object> config = new LinkedHashMap<>();
}

File: examples/kotlin/core/shared/app_base_context.kt
package app.protocol.examples.kotlin.core.shared

import app.protocol.examples.kotlin.core.Dict

interface AppLogger {
    fun debug(message: String, meta: Dict<Any?>? = null)

    fun info(message: String, meta: Dict<Any?>? = null)

    fun warn(message: String, meta: Dict<Any?>? = null)

    fun error(message: String, meta: Dict<Any?>? = null)
}

interface AppBaseContext {
    val correlationId: String
    val executionId: String?
    val tenantId: String?
    val userId: String?
    val logger: AppLogger
    val config: Dict<Any?>?
}
